### PR TITLE
androidmedia: Add support for HEVC and VP8/9

### PR DESCRIPTION
**Pull request by alfonsosanchezbeato**
Add capability to decode HEVC and VP8/9 streams in case the android HAL supports them.

### CODE REVIEW

File: sys/androidmedia/gstamc-constants.h [hunk 1]
```diff
@@ -101,7 +101,10 @@ enum
   /* From hardware/ti/omap4xxx/domx/omx_core/inc/OMX_TI_IVCommon.h */
   COLOR_TI_FormatYUV420PackedSemiPlanarInterlaced = 0x7f000001,
   COLOR_EXYNOS_FormatNV12Tiled = 0x7fc00002,
-  COLOR_EXYNOS_FormatNV21Linear = 0x7f000011
+  COLOR_EXYNOS_FormatNV21Linear = 0x7f000011,
+  /* MTK formats */
```
Comment (jhodapp) [inline sys/androidmedia/gstamc-constants.h]: Further explain what these colors are for, that they were added for HEVC (h.265) and VP8/9 support. Can you explicitly label which one is for which format?

File: sys/androidmedia/gstamchybris.c [hunk 2]
```diff
@@ -1186,11 +1186,6 @@ scan_codecs (GstPlugin * plugin)
 
       n_elems = media_codec_list_get_num_profile_levels (i, mime);
       GST_INFO ("Type '%s' has %d supported profile levels", mime, n_elems);
-      if (n_elems == 0) {
```
Comment (jhodapp) [inline sys/androidmedia/gstamchybris.c]: Why was this removed?

File: sys/androidmedia/gstamcvideodechybris.c [hunk 3]
```diff
@@ -1750,6 +1793,39 @@ gst_amc_video_dec_flush (GstVideoDecoder * decoder)
   return TRUE;
 }
 
+static unsigned
+skip_forbidden_nalus(const GstAmcVideoDec *self, GstMapInfo *minfo,
```
Comment (jhodapp) [inline sys/androidmedia/gstamcvideodechybris.c]: Shouldn't this be skip_forbidden_nals? What is the 'u' for?

File: sys/androidmedia/gstamcvideodechybris.c [hunk 4]
```diff
@@ -1750,6 +1793,39 @@ gst_amc_video_dec_flush (GstVideoDecoder * decoder)
...
+   * detect it we search for type 9 NALU after a start code (0x00000001).
+   */
+  if (g_strcmp0("video/avc", self->mime) == 0 &&
+      minfo->size - offset > 5 &&
+      GST_READ_UINT32_BE (minfo->data + offset) == 0x01 &&
+    (*(minfo->data + offset + 4) & 0x1F) == 9) {
+    offset += 6;
+  }
+
+  /* The VPS nalus introduced in HEVC are not handled by MTK decoder, we move
```
Comment (jhodapp) [inline sys/androidmedia/gstamcvideodechybris.c]: What does VPS mean here, can you expand this acronym?

File: sys/androidmedia/gstamc-constants.h (hunk 1 above)
Comment (alfonsosanchezbeato) [inline sys/androidmedia/gstamc-constants.h]: These 2 formats have been added to OMX_COLOR_FORMATTYPE (OMX_IVCommon.h) by the SoC vendor. FormatYV12 is one of the color formats used by VP8/9 in arale. The other one is added for completeness. I'll add a comment in the code.

Comment (jhodapp) [inline sys/androidmedia/gstamc-constants.h]: Excellent, thanks.

File: sys/androidmedia/gstamchybris.c (hunk 2 above)
Comment (alfonsosanchezbeato) [inline sys/androidmedia/gstamchybris.c]: VP8 does not have profiles, so this line was preventing its registration as an androidmedia plugin. No regressions exist as this simply was putting more restrictions on which codecs we considered valid. When the gstreamer registry is re-created, the previous codecs are created with the same properties as before.

File: sys/androidmedia/gstamcvideodechybris.c (hunk 4 above)
Comment (alfonsosanchezbeato) [inline sys/androidmedia/gstamcvideodechybris.c]: Comment added in the code.

**Comment by alfonsosanchezbeato** [comment]
Re-pulled addressing issues.
